app/components/file/browse/blob-files-browser/blob-files-browser.component.ts

```diff
@@ -52,11 +57,40 @@ export class BlobFilesBrowserComponent implements OnChanges, OnDestroy {
                 this.upload(event).subscribe(() => {
                     this.fileNavigator.refresh(path);
                 });
+
                 return Observable.of(null);
             },
         });
     }
 
+    public handleDeleteEvent(event: FileDeleteEvent) {
+        const { path } = event;
+        console.log("BlobFilesBrowserComponent.handleDeleteEvent: ", event);
+
+        // TODO: Check with Tim as to whether i can chain 2 background tasks to run one after another.
+        // In which case most of this can be moved into DataContainerFilesComponent.
+        // task 1: get files to delete
+        // task 2: delete files
```

Comment:
you could but I don't think thats a good idea, it would look weird. Maybe what you could do is have it send back an observable that resolve files instead of the actual files so the waiting can be done in the background task

Reply:
but otherwise I think its all good like that for now until we make that generic FileSystem